src/main/java/com/spencerwi/either/Either.java

```diff
@@ -41,6 +41,22 @@
 
     public abstract <R2> Either<L, R2> flatMapRight(Function<R, Either<L,R2>> transformRight);
 
+    public abstract L getLeftOrElse(L other);
+
+    public abstract L getLeftOrElse(Supplier<L> otherSupplier);
+
+    public abstract <X extends Throwable> L getLeftOrElseThrow(Function<R,X> exceptionFunction) throws X;
```

Comment:
This signature is honestly just a bit confusing. I think we could drop this one (and the corresponding `getRightOrElseThrow` overload), and the usage would be clearer.